src/main/kotlin/no/skatteetaten/aurora/gobo/integration/imageregistry/ImageRegistryServiceBlocking.kt

```diff
@@ -1,114 +1,99 @@
     }
 
     fun findTagNamesInRepoOrderedByCreatedDateDesc(imageRepoDto: ImageRepoDto): List<String> {
-
         val registryMetadata = registryMetadataResolver.getMetadataForRegistry(imageRepoDto.registry)
 
-        val tagListUrl = urlBuilder.createTagListUrl(registryMetadata.apiSchema, imageRepoDto)
-
-        val tagList: TagList? = getTags(tagListUrl, registryMetadata.authenticationMethod)
-        val tagsOrderedByCreatedDate = tagList?.tags ?: emptyList()
+        val tagsOrderedByCreatedDate: List<String>? = getTags(imageRepoDto, registryMetadata.authenticationMethod)
 
         // The current image registry returns the tag names in the order they were created. There does not, however,
         // seem to be a way to affect what property the tags are ordered by or the direction they are ordered in, so
         // there is a chance this order is an "undocumented feature" of the api. We will rely on this feature for the
         // time being, though, as it allows for some queries to be significantly quicker than fetching the individual
         // created dates for each tag, and then sort.
-        return tagsOrderedByCreatedDate.reversed()
+        return tagsOrderedByCreatedDate?.reversed() ?: emptyList()
     }
 
     private fun getImageMetaData(imageRepoDto: ImageRepoDto, tag: String): ImageMetadata? {
-
         val registryMetadata = registryMetadataResolver.getMetadataForRegistry(imageRepoDto.registry)
 
         return try {
-            val manifestsUrl = urlBuilder.createManifestsUrl(registryMetadata.apiSchema, imageRepoDto, tag)
-            getManifest(manifestsUrl, registryMetadata.authenticationMethod)
-                ?.let { parseManifest(it) }
+            getManifestInformation(imageRepoDto, tag, registryMetadata.authenticationMethod)
         } catch (e: Exception) {
             throw SourceSystemException("Unable to get manifest for image: $tag", e)
         }
     }
 
-    private fun parseManifest(manifestString: String): ImageMetadata {
-
-        val manifest = objectMapper.readTree(manifestString)
-        val manifestHistory = manifest.get("history").get(0).get("v1Compatibility")
-        val manifestFirstHistory = objectMapper.readTree(manifestHistory.asText())
-        val createdString = manifestFirstHistory.get("created").asText()
-        return ImageMetadata(Instant.parse(createdString))
-    }
-
-    private fun getTags(apiUrl: String, authenticationMethod: AuthenticationMethod): TagList? {
-        return getFromRegistry(apiUrl, authenticationMethod)
-    }
-
-    private fun getManifest(apiUrl: String, authenticationMethod: AuthenticationMethod): String? {
-        return getFromRegistry(apiUrl, authenticationMethod)
-    }
+    private fun getManifestInformation(
+        imageRepoDto: ImageRepoDto,
+        imageTag: String,
+        authenticationMethod: AuthenticationMethod
+    ): ImageMetadata? =
+        getFromCantus(
```

Comment:
Hvis du skal få gode metrikker på utgående http kall bør disse urlene være formatert med placeholdere i url kallet i webClient for cantus.

Reply:
se f.eks getApplicationDeployment i ApplicationService

Reply:
Fikset :)